src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+        Math_SmoothStepToS(&this->picto.actor.shape.rot.z, this->picto.actor.world.rot.z, 5, 0x3000, 0x100);
+    } else if (this->timer < 30) {
+        this->timer++;
+        this->picto.actor.shape.rot.y = (s32)(Math_CosS(this->skyFloatPhase) * 2500.0f) + this->picto.actor.world.rot.y;
+        this->picto.actor.shape.rot.z = (s32)(Math_SinS(this->skyFloatPhase) * 2500.0f) + this->picto.actor.world.rot.z;
+        if (this->timer == 25) {
+            this->forceEyesShut = false;
+            this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_FALL_ONCE);
+        } else if (this->timer == 29) {
+            this->picto.actor.velocity.y = -12.0f;
+        }
+    }
+
+    if (this->picto.actor.bgCheckFlags & BGCHECKFLAG_GROUND_TOUCH) {
+        Actor_PlaySfx(&this->picto.actor, NA_SE_VO_TIVO01);
+        Actor_PlaySfx(&this->picto.actor, NA_SE_EN_GERUDOFT_DOWN);
+    }
+
+    if (this->picto.actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
+        this->picto.actor.colChkInfo.health = 0;
+        if (this->timer > 80) {
+            this->picto.actor.shape.yOffset = 0.0f;
+            EnBal_SetupGroundIdle(this);
+        } else if (this->timer == 30) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_LAND);
+            this->picto.actor.shape.rot = this->picto.actor.world.rot;
+            Actor_SpawnFloorDustRing(play, &this->picto.actor, &worldPos, 10.0f, 30, 5.0f, 0, 0, 0);
+            this->timer++;
+        } else {
+            if (play->gameplayFrames & 1) {
+                this->picto.actor.shape.yOffset = 20.0f;
+            } else {
+                this->picto.actor.shape.yOffset = 0.0f;
+            }
+            this->timer++;
+        }
+    } else if (this->picto.actor.velocity.y < 0.0f) {
+        Actor_PlaySfx_Flagged(&this->picto.actor, NA_SE_EV_HONEYCOMB_FALL - SFX_FLAG);
+    }
+
+    this->skyFloatPhase += 3000;
+}
+
+void EnBal_SetupInflateBalloon(EnBal* this) {
+    Vec3f scale = { 0.0f, 0.0f, 0.0f };
+
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_FLOAT_DRAW);
+    this->prevBalloonAction = EN_BAL_BALACT_INFLATE;
+    this->watchTarget = EN_BAL_WATCHTRG_NONE;
+    this->forceEyesShut = false;
+    this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+    this->balloonScale = scale;
+    this->skyFloatPhase = 0;
+    this->timer = 0;
+    this->actionFunc = EnBal_InflateBalloon;
+}
+
+void EnBal_InflateBalloon(EnBal* this, PlayState* play) {
+    f32 scale;
+
+    if (Math_SmoothStepToF(&this->balloonScale.x, 1.0f, 0.05f, 0.03f, 0.01f) == 0.0f) {
+        EnBal_SetupFloatUpToSky(this);
+    }
+    EnBal_EmitDustPuff(this, play);
+    Actor_PlaySfx_Flagged(&this->picto.actor, NA_SE_EV_BALLOON_SWELL - SFX_FLAG);
+    scale = this->balloonScale.x * 1.1f;
+    this->balloonScale.z = scale;
+    this->balloonScale.y = scale;
+}
+
+void EnBal_SetupFloatUpToSky(EnBal* this) {
```

Comment:
```suggestion
void EnBal_SetupFloatUp(EnBal* this) {
```
Again `ToSky` doesn't seem to add much.